--- Makefile.PL.orig	Tue Apr 16 13:43:17 2002
+++ Makefile.PL	Tue Apr 16 13:43:35 2002
@@ -30,8 +30,6 @@
 print "Config directory: $confdir\n";
 $cc=`$apxs -q CC`;
 print "Compiler: $cc\n";
-$cflags=`$apxs -q CFLAGS`;
-print "CFLAGS: $cflags\n";
 $includedir=`$apxs -q INCLUDEDIR`;
 print "Include directory: $includedir\n";
 $libexecdir=`$apxs -q LIBEXECDIR`;
@@ -105,7 +103,6 @@
 $_=~ s|\$\(apachectl\)|$apachectl|;
 $_=~ s|\$\(cc\)|$cc|;
 $_=~ s|\$\(includedir\)|$includedir|;
-$_=~ s|\$\(cflags\)|$cflags|;
 $_=~ s|\$\(sbindir\)|$sbindir|;
 $_=~ s|\$\(install\)|$install|;
 #$_=~ s|\$\(\)|$|;
