--- fpexec.h.orig	Sun Feb  3 16:31:52 2002
+++ fpexec.h	Tue Apr 16 14:25:20 2002
@@ -69,7 +69,7 @@
  *                  this program.
  */
 #ifndef FP_HTTPD_USER
-#define FP_HTTPD_USER "www"
+#define FP_HTTPD_USER "http"
 #endif
 
 /*
@@ -165,7 +165,7 @@
  *               that can be used for fpEXEC behavior.
  */
 #ifndef FP_DOC_ROOT
-#define FP_DOC_ROOT "/usr/local/apache/htdocs"
+#define FP_DOC_ROOT "/home/httpd/html"
 #endif
 
 /*
